xds/src/test/java/io/grpc/xds/FilterChainMatchingProtocolNegotiatorsTest.java

```diff
@@ -0,0 +1,1318 @@
+    assertThat(sslSet.isDone()).isTrue();
+    assertThat(sslSet.get()).isEqualTo(f0.getSslContextProviderSupplier());
+    channelHandlerCtx = pipeline.context(next);
+    assertThat(channelHandlerCtx).isNotNull();
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void nofilterChainMatch_defaultSslContext() throws Exception {
+    final SettableFuture<SslContextProviderSupplier> sslSet = SettableFuture.create();
+    ChannelHandler next = new ChannelInboundHandlerAdapter() {
+      @Override
+      public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
+        ProtocolNegotiationEvent e = (ProtocolNegotiationEvent)evt;
+        sslSet.set(InternalProtocolNegotiationEvent.getAttributes(e)
+                .get(ATTR_SERVER_SSL_CONTEXT_PROVIDER_SUPPLIER));
+      }
+    };
+    when(mockDelegate.newHandler(grpcHandler)).thenReturn(next);
+
+    SslContextProviderSupplier ssl = new SslContextProviderSupplier(createTls(), tlsContextManager);
+    FilterChainSelector selector = new FilterChainSelector(Collections.EMPTY_LIST, ssl);
+    FilterChainMatchingHandler filterChainMatchingHandler =
+            new FilterChainMatchingHandler(grpcHandler, selector, mockDelegate);
+    setupChannel("172.168.1.1", "172.168.1.2", 80, filterChainMatchingHandler);
+    ChannelHandlerContext channelHandlerCtx = pipeline.context(filterChainMatchingHandler);
+    assertThat(channelHandlerCtx).isNotNull();
+
+    pipeline.fireUserEventTriggered(event);
+    channelHandlerCtx = pipeline.context(filterChainMatchingHandler);
+    assertThat(channelHandlerCtx).isNull();
+
+    channel.runPendingTasks();
+    assertThat(sslSet.isDone()).isTrue();
+    assertThat(sslSet.get()).isEqualTo(ssl);
+    channelHandlerCtx = pipeline.context(next);
+    assertThat(channelHandlerCtx).isNotNull();
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void noFilterChainMatch_noDefaultSslContext() {
+    FilterChainSelector selector = new FilterChainSelector(Collections.EMPTY_LIST, null);
+    FilterChainMatchingHandler filterChainMatchingHandler =
+            new FilterChainMatchingHandler(grpcHandler, selector, mockDelegate);
+    setupChannel("172.168.1.1", "172.168.2.2", 90, filterChainMatchingHandler);
+    channelHandlerCtx = pipeline.context(filterChainMatchingHandler);
+    assertThat(channelHandlerCtx).isNotNull();
+
+    pipeline.fireUserEventTriggered(event);
+    channelHandlerCtx = pipeline.context(filterChainMatchingHandler);
+    assertThat(channelHandlerCtx).isNotNull();
+    try {
+      channel.checkException();
+      fail("exception expected!");
+    } catch (Exception e) {
+      assertThat(e).isInstanceOf(IllegalStateException.class);
+      assertThat(e).hasMessageThat().contains("No matching filter chain found.");
+    }
+  }
+
+  @Test
+  public void singleFilterChainWithoutAlpn() throws Exception {
+    EnvoyServerProtoData.FilterChainMatch filterChainMatch =
+            new EnvoyServerProtoData.FilterChainMatch(
```

Comment:
The indentation in many places in this file is 8 instead of the default 4. You can format using Google-java-format plugin in your IDE.

Reply:
Is it a blocker? checkstyle doesnt complain.
I am not using glinux and it looks Google-java-format/google vim is not supporting mac well. It looks tedious into them for now.